### PR TITLE
Cleanup TokenizedPhraseQueryNode code

### PR DESCRIPTION
**Pull request by sabi0**
- harmonize implementations of the methods working with children
- save `getChildren()` in a variable and re-use it
- use more efficient `isEmpty()` instead of `size() == 0`
- fix a typo in the tag name: `tokenized_t_phrase` => `tokenizedphrase`
- remove unnecessary `toString()` call
- remove trailing comment referring to another class (copy paste? renaming that did not update the comment?)

### CODE REVIEW

File: lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java [hunk 1]
```diff
@@ -70,10 +70,8 @@ public QueryNode cloneTree() throws CloneNotSupportedException {
   @Override
   public CharSequence getField() {
     List<QueryNode> children = getChildren();
-
-    if (children == null || children.size() == 0) {
+    if (children == null || children.isEmpty()) {
       return null;
-
     } else {
       return ((FieldableNode) children.get(0)).getField();
```
Comment (sabi0) [inline lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java:76]: According to the `setField(CharSequence)` method below children are not necessarily `FieldableNode`.
I.e. this line might throw `ClassCastException`?

Comment (stefanvodita) [reply lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java:76]: I see your point. Maybe it's worth adding an assertion?

**Review by stefanvodita — APPROVED** [review]
The clean-up looks good. Sorry it took so long to get a review.

File: lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java (hunk 1 above)
Comment (sabi0) [reply lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java:76]: I think the two methods should be aligned.
If one has an assert (assumes all the children are `FieldableNode`) then the other should do so too.
Of they both should silently ignore children of other types.

I would probably go with "silently ignore children of other types" as the least surprising option.

Comment (stefanvodita) [reply lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java]: I think having the two methods aligned, the way they are in your latest commit, makes sense.

File: lucene/CHANGES.txt [hunk 2]
```diff
@@ -258,6 +258,8 @@ Improvements
   Tests are running with random byte order to ensure that the order does not affect correctness
   of code. Native order was enabled for LZ4 compression.  (Uwe Schindler)
 
+* GITHUB#13041: TokenizedPhraseQueryNode code cleanup (Dmitry Cherniachenko)
```
Comment (stefanvodita) [inline lucene/CHANGES.txt:261]: This would have to move up to version 9.11 because 9.10 is already in the release process.

**Pull request merged**